File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
    
    bool neg = x<0;
 

    long long num = abs((long long)x);

    string str = to_string(num);
    std::reverse(str.begin(), str.end());

    long long ans = stoll(str);

    if(neg) ans = -ans;

    if (ans > INT_MAX || ans < INT_MIN)
            return 0;
    return ans;
     
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        boolean neg = true;
        if (x < 0)
            neg = false;

        long num = Math.abs((long) x);

        String str = String.valueOf(num);
        String reverse = "";

        for (int i = str.length() - 1; i >= 0; i--) {

            reverse += str.charAt(i);
        }

        long res = Long.parseLong(reverse);

        if (neg == false)
            res = -res;

        if (res > Integer.MAX_VALUE || res < Integer.MIN_VALUE) {
            return 0;
        } else {
            return (int)res;
        }

    }
}